TFNetworkLayer.py

```diff
@@ -2970,14 +2981,15 @@ def _get_out_type_from_opts(cls, n_out, filter_size, padding, strides=1, dilatio
     padding = padding.upper()
     if input_expand_dims == 0 and not input_add_feature_dim and not input_split_feature_dim:
       # Maybe we have a chance to correctly define the output shapes.
-      data = get_concat_sources_data_template(sources)
+      index_shift = data.time_dim_axis_excluding_batch
       for i in range(len(filter_size)):
-        if data.shape[i] is not None:
-          shape[i] = cls.calc_out_dim(
-            in_dim=data.shape[i],
+        if data.shape[i + index_shift] is not None:
+          shape[i + index_shift] = cls.calc_out_dim(
```

Comment:
Follow my comment above (just leave `shape = [None] * len(filter_size) + [n_out]`).
Then here you must have `shape[i] = cls.calc_out_dim(...`.